serializations.md

```diff
@@ -1029,3 +1029,539 @@ NOTE:
 , <pubkey>   :: binary()
 ]
 ```
+
+## Serialization of FATE contracts
+
+FATE contracts uses a recursive BNF like definition.
+There are serialisations (RLP, FATE), sequences of serializations (`S1, S2, ...`),
+sequences of bytes (`<< .... >>`) and sequences of bits (`<<< ... >>>`).
+
+Fate Code uses both RLP enodings and pure byte arrays (or binary) encodings.
+In the description here we write `RLP(X)` for the RLP encoding of `X`, where
+X is a name of an encoding described above or below in the description of
+the FATE encoding. We write `<X, Y>` for a byte array made of the sequence of
+the byte encodings of `X` and `Y`. We write `X(Y)` to denote the encoding
+of the type `Y` using the encoding `X`.
+
+We write `< >` for the empty sequence and use `|` for alternatives. We
+use hexadecimal numbers to represent specific bytes used to indicate
+what is encoded, e.g. `0xfe` for a function. We use `_` to denote any
+byte in a byte sequence `<< _ >>` or any encoding in a encoding
+sequence ` _, `.
+
+We use `<<< X, Y >>>` to denote the bitpattern of the bits `X` followed
+by the bits `Y` where the bits on the left (`X`) are the most significant bits.
+A bitpattern should produce a number of bits devisable by 8, if not the pattern
+should be padded with zeroes on the left (in the high order) until it is devisable by 8.
+We use ```<<< xy >>>``` as a shorthand for ```<<< x, y >>>``` when `x` and `y` are
+just 0 or 1 and not a recursive definition.
+
+
+We use `; Text` to denote a comment that is not part of the encoding but
+used to make the meaning of the encoding clear.
+
+### FATE top level
+A fate contract is serialized as follows
+```
+Contract ::=
+  RLP(Code),
+  RLP(Symbols),
```

Comment:
I don't see Symbols and Annotations explained 🤔

Reply:
Annotations are not used AFAIK, Symbols should probably be explained, but I'd have to read the code to do it 🙄